tests/schema/test_fields_default.py

```diff
@@ -0,0 +1,101 @@
+import strawberry
+
+
+def test_default_value():
+    @strawberry.type
+    class Query:
+        a: str = strawberry.field(default="Example")
+
+    schema = strawberry.Schema(query=Query)
+
+    query = "{ a }"
+
+    result = schema.execute_sync(query, root_value=Query())
+
+    assert not result.errors
+    assert result.data == {"a": "Example"}
+
+
+def test_default_factory():
+    @strawberry.type
+    class Query:
+        a: str = strawberry.field(default_factory=lambda: "Example")
+
+    schema = strawberry.Schema(query=Query)
+
+    query = "{ a }"
+
+    result = schema.execute_sync(query, root_value=Query())
+
+    assert not result.errors
+    assert result.data == {"a": "Example"}
+
+
+def test_default_factory_multiple_fields():
+    # NOTE: The default_factory is called when instantiating the schema
+    # this is why the user will start from 2
+    counter = 0
+
+    def generate_name():
+        nonlocal counter
+        counter += 1
+        return f"User {counter}"
+
+    @strawberry.type
+    class User:
+        name: str = strawberry.field(default_factory=generate_name)
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def user(self) -> User:
+            return User()
+
+        @strawberry.field
+        def user2(self) -> User:
+            return User()
+
+    schema = strawberry.Schema(query=Query)
+
+    query = "{ user { name }, user2 { name } }"
+
+    result = schema.execute_sync(query, root_value=Query())
+
+    assert not result.errors
+    assert result.data == {"user": {"name": "User 2"}, "user2": {"name": "User 3"}}
+
+
+def test_default_factory_input():
```

Comment:
@patrick91 
How is this test related to default enums? :thinking:

Reply:
Anyways maybe some caching is preventing the factory from being used in `update2`

Reply:
🤔

Reply:
that was a file I added by mistake, I was working on something else :D